src/terraformpy/objects.py

```diff
@@ -54,7 +54,7 @@ class DuplicateKey(str):
     def __new__(cls, key):
         inst = super(DuplicateKey, cls).__new__(cls, key)
 
-        inst._hash = DuplicateKey._next_hash[key]
+        inst._hash = hash((key, DuplicateKey._next_hash[key]))
```

Comment:
Per the Python docs:

> it is advised to mix together the hash values of the components of the object that also play a part in comparison of objects by packing them into a tuple and hashing the tuple.

https://docs.python.org/3/reference/datamodel.html#object.__hash__